src/main/java/edu/harvard/dbmi/avillach/dictionary/concept/ConceptFilterQueryGenerator.java

```diff
@@ -17,9 +17,13 @@ public class ConceptFilterQueryGenerator {
 
     private static final String CONSENT_QUERY = """
                 dataset.dataset_id IN (
-                    SELECT consent.dataset_id
+                    SELECT
+                        consent.dataset_id
                     FROM consent
-                    WHERE consent.consent_code IN (:consents)
+                        LEFT JOIN dataset ON dataset.dataset_id = consent.dataset_id
+                    WHERE
+                        concat(dataset.ref, '.', consent.consent_code) IN (:consents) OR
+                        concat(dataset.ref) IN (:consents)
```

Comment:
This isnt doing any partial-matching right? Like if someone has access to phs1 c1 but not phs c2, the consents list will have phs1.c1 in the list, and phs1.c2 shouldnt be matched by the check that is only dataset

Reply:
correct. IN is equality

Reply:
oh I see what you mean
code needs to be empty
good shit

Reply:
fixed